CountedSet.playground/Contents.swift

```diff
@@ -0,0 +1,58 @@
+import UIKit
+
+
+struct CountedSet<Element: Hashable>: ExpressibleByArrayLiteral {
+    
+    
+    var dict: [Element: Int] = [:]
+    
+    
+    init(arrayLiteral elements: Element...) {
+        for element in elements {
+            insert(element)
+        }
+    }
+    
+    mutating func insert(_ element: Element) {
+        dict[element] = (dict[element] ?? 0) + 1
+    }
+    
+    mutating func remove(_ element: Element) -> Int{
+        var count = dict[element] ?? 0
+        
+        if count > 0 {
+            dict[element] = count - 1
+            count -= 1
+        }
```

Comment:
So if the count of an element is 1, it gets decremented to 0. The only issue with that is that the element never actually gets removed from the counted set, it's count just gets set to 0.